src/problems/basic_problems.jl

```diff
@@ -391,6 +395,56 @@ struct QuadratureProblem end
 
 @doc doc"""
 
+Defines a integral problem over pre-sampled data.
+Documentation Page: https://docs.sciml.ai/Integrals/stable/
+
+## Mathematical Specification of a data Integral Problem
+
+Sampled integral problems are defined as:
+
+```math
+\sum_i w_i y_i
+```
+where `y_i` are sampled values of the integrand, and `w_i` are weights 
+assigned by a quadrature rule, which depend on sampling points `x`. 
+
+## Problem Type
+
+### Constructors
+
+```
+SampledIntegralProblem(x::AbstractVector, y::AbstractArray; dim=1, kwargs...)
+```
+- y: The sampled integrand, must be a subtype of `AbstractArray`. 
+  It is assumed that the values of `y` along dimension `dim` 
+  correspond to the integrand evaluated at sampling points `x`
+- x: Sampling points, must be a subtype of `AbstractVector`.   
+- dim: Dimension along which to integrate.
```

Comment:
Is there a reason to leave it up to the user? I think that just adds more work in the implementation without a tangible benefit. dimension should always be the last one as other wise you'll have a performance hit from column major, so we should just stick to that.

Reply:
I disagree, with both (1) that the last dim is always faster, and (2) even if it were that we should force all users to use that.

1. Here is a simple example:
 ```
function trapz(y, dim)
           solve(SampledIntegralProblem(y, axes(y,dim); dim=dim), 
            TrapezoidalRule())
end
y = rand(10000, 5)
julia> @btime trapz(trapz($y,2), 1)
  117.700 μs (8 allocations: 400 bytes)
u: 19952.961100082917
julia> @btime trapz(trapz($y,1), 1)
  37.900 μs (9 allocations: 78.47 KiB)
u: 19952.961100082754
```
  while there are better ways to do a two-dimensional trapezoidal rule of course.

2. Usually, doing numeric integration over some array isn't the performance bottleneck of the algorithm. Typically, generating the data in the first place (by some experiment perhaps) takes more time. It would be annoying to force users to stick to a certain data layout to be able to do integration over it if it is just a small part of their workflow. This is also why other packages implement this feature, see SciPy or Trapz.jl for example.

Having said that, you are of course right that the last dimension is typically faster and should be the default, so I'll change that.